uvm/bootstrap/com/untangle/uvm/UvmContextSelector.java

```diff
@@ -0,0 +1,234 @@
+/**
+ * $Id$
+ */
+
+package com.untangle.uvm;
+
+import org.apache.logging.log4j.core.Appender;
+import org.apache.logging.log4j.core.LoggerContext;
+import org.apache.logging.log4j.core.appender.SyslogAppender;
+import org.apache.logging.log4j.core.config.Configuration;
+import org.apache.logging.log4j.core.config.ConfigurationSource;
+import org.apache.logging.log4j.core.config.LoggerConfig;
+import org.apache.logging.log4j.core.config.xml.XmlConfiguration;
+import org.apache.logging.log4j.core.layout.PatternLayout;
+import org.apache.logging.log4j.core.net.Facility;
+import org.apache.logging.log4j.core.net.Protocol;
+import org.apache.logging.log4j.core.selector.ContextSelector;
+import org.apache.logging.log4j.status.StatusLogger;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Custom ContextSelector implementation to create multiple logger context for apps
+ */
+public class UvmContextSelector implements ContextSelector {
+
+    private final ConcurrentHashMap<String, LoggerContext> contexts = new ConcurrentHashMap<>();
+    private static final StatusLogger LOGGER = StatusLogger.getLogger();
+    public static final String UVM_LOG = "uvm";
+    private static final UvmContextSelector INSTANCE;
+    private static final ThreadLocal<String> THREAD_LOG_INFO = new InheritableThreadLocal<>();
+
+    static {
+        INSTANCE = new UvmContextSelector();
+    }
+
+    /**
+     * instance() Provides the UvmContextSelector singleton
+     * @return UvmContextSelector
+     */
+    public static UvmContextSelector instance() {
+        return INSTANCE;
+    }
+
+    /**
+     * Overrides getContext method of ContextSelector interface
+     * @param fqcn fqcn
+     * @param loader loader
+     * @param currentContext boolean isCurrentContext
+     * @return LoggerContext
+     */
+    @Override
+    public LoggerContext getContext(String fqcn, ClassLoader loader, boolean currentContext) {
+        try{
+            String contextName = THREAD_LOG_INFO.get();
```

Comment:
In case we encountered any exception while loading configuration, we are returning  `contexts.get(UVM_LOG);`
`UVM_LOG` in current code and `DEFAULT_LOG` in old code refers to same file i.e. `uvm`